src/Native/Bootstrap/main.cpp

```diff
@@ -372,6 +372,11 @@ extern "C" void RhpBulkWriteBarrier()
 {
     throw 42;
 }
+
+extern "C" Object* RhNewArrayAsString(MethodTable*)
```

Comment:
Here is the other implementation that pulls in the dependency:

extern "C" System_Private_CoreLib::System::String* RhNewArrayAsString(System_Private_CoreLib::System::EETypePtr, int32_t);
System_Private_CoreLib::System::String* System_Private_CoreLib::System::Runtime::RuntimeImports::RhNewArrayAsString(System_Private_CoreLib::System::EETypePtr pEEType, int32_t length)
{
    return ::RhNewArrayAsString(pEEType, length);
}